src/main/java/unibook/logic/commands/EditCommand.java

```diff
@@ -57,6 +89,113 @@ public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
 
         this.index = index;
         this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+        this.editModuleDescriptor = null;
+    }
+
+    /**
+     * @param index of the module in the filtered module list to edit
+     * @param editModuleDescriptor details to edit the module with
+     */
+    public EditCommand(Index index, EditModuleDescriptor editModuleDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editModuleDescriptor);
+
+        this.index = index;
+        this.editModuleDescriptor = new EditModuleDescriptor(editModuleDescriptor);
+        this.editPersonDescriptor = null;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException, ModuleNotFoundException {
+        requireNonNull(model);
+
+        if (this.editModuleDescriptor == null) {
+            List<Person> lastShownList = model.getFilteredPersonList();
+            List<Module> latestModList = model.getFilteredModuleList();
+
+            if (index.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+
+            Person personToEdit = lastShownList.get(index.getZeroBased());
+            Module checkMod = null;
+            if (!editPersonDescriptor.getModules().equals(Optional.empty()) && latestModList.size() != 0) {
+                checkMod = editPersonDescriptor.getModules().get().iterator().next();
+                if (!latestModList.contains(checkMod)) {
+                    throw new ModuleNotFoundException();
+                }
+            }
+
+            Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+
+            // Temporarily using tags to identify whether person is prof or student
+            // Need to change the EditPersonDescriptor to include type of person, maybe EditProfDescriptor etc
+            String editedPersonType = null;
+
+            // When adding new module with nm/, adds prof/student to person list in each mod
+            if (checkMod != null) {
+                editedPersonType = editPersonDescriptor.getTags().get().iterator().next().tagName;
+                int modIdx = latestModList.indexOf(checkMod);
+                if (editedPersonType.equals("professor")) {
+                    latestModList.get(modIdx).addProfessor((Professor) editedPerson);
+                } else {
+                    latestModList.get(modIdx).addStudent((Student) editedPerson);
```

Comment:
To add on here, it could maybe not be ideal to use `addStudent`, as this always appends to the back of the list instead of preserving the original position of the student. For example if we edit student 3 in a list of 7 students, the edited person always goes to the back. Maybe could implement a `setStudent` method to allow editing at a specific index?